WalletWasabi/WabiSabi/Client/CoinJoinManager.cs

```diff
@@ -237,8 +237,8 @@ private async Task MonitorAndHandlingCoinJoinFinallizationAsync(ConcurrentDictio
 			var finishedCoinJoins = trackedCoinJoins.Where(x => x.Value.IsCompleted).Select(x => x.Value).ToImmutableArray();
 			foreach (var finishedCoinJoin in finishedCoinJoins)
 			{
-				NotifyCoinJoinCompletion(finishedCoinJoin);
 				await HandleCoinJoinFinalizationAsync(finishedCoinJoin, trackedCoinJoins, stoppingToken).ConfigureAwait(false);
+				NotifyCoinJoinCompletion(finishedCoinJoin);
```

Comment:
Reverse the order of call. If we notify before we finish, coins are not refrigerated and another CoinJoin client might be able to use them.